protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArray.java

```diff
@@ -200,8 +218,11 @@ public int getTotalTrackedNodeCount() {
    * @param finalizedRoot
    */
   public void maybePrune(Bytes32 finalizedRoot) {
-    int finalizedIndex =
-        checkNotNull(indices.get(finalizedRoot), "ProtoArray: Finalized root is unknown");
+    Optional<Integer> maybeIndex = protoArrayIndices.getFirst(finalizedRoot);
+    checkArgument(
+        maybeIndex.isPresent(),
+        "ProtoArray: Finalized root is unknown " + finalizedRoot.toHexString());
+    int finalizedIndex = maybeIndex.get();
```

Comment:
Maybe just:
```suggestion
    int finalizedIndex = protoArrayIndices.getFirst(finalizedRoot).orElseThrow(() -> new IllegalArgumentException("ProtoArray: Finalized root is unknown " + finalizedRoot.toHexString());
```

(I guarantee spotless won't like that formatting)